go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -588,7 +589,36 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		}
 	}
 
-	ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bnc.nbClient, nil, ls, lsp)
+	lsps := []*nbdb.LogicalSwitchPort{lsp}
+	if kubevirt.IsPodOwnedByVirtualMachine(pod) && isAllowedForMigration(bnc.IsSecondary(), bnc.IsPrimaryNetwork(), bnc.isLayer2Interconnect()) {
+		kubevirtLiveMigrationStatus, err := kubevirt.DiscoverLiveMigrationStatus(bnc.watchFactory, pod)
```

Comment:
I don't like that we are adding this to the `base_network_controller_pods` since it's executed with ALLLL the pods, can you try to move the call to Discovery at https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/base_network_controller_secondary.go#L305-L309 ? 

At the end is just about adding another "ops" and they get commited like a transaction later on.

Reply:
moving it there means it still in both secondary+primary UDN code path right?

Reply:
Yep, everything that is UDN, proper place would be the controller that is common to localnet and layer2 `BaseSecondaryLayer2NetworkController` but don't know if it's worth it or possible with some refactoring.